--- AppDelegate.cpp	2014-12-12 21:47:28.000000000 +0800
+++ AppDelegate.cpp	2015-09-10 16:55:08.000000000 +0800
@@ -4,6 +4,13 @@
 #include "cocos2d.h"
 #include "lua_module_register.h"
 
+#if (CC_TARGET_PLATFORM == CC_PLATFORM_ANDROID || CC_TARGET_PLATFORM == CC_PLATFORM_IOS)
+#include "AgentManager.h"
+#include "anysdkbindings.h"
+#include "anysdk_manual_bindings.h"
+using namespace anysdk::framework;
+#endif
+
 using namespace CocosDenshion;
 
 USING_NS_CC;
@@ -16,6 +23,10 @@
 AppDelegate::~AppDelegate()
 {
     SimpleAudioEngine::end();
+        
+#if (CC_TARGET_PLATFORM == CC_PLATFORM_ANDROID || CC_TARGET_PLATFORM == CC_PLATFORM_IOS)
+    AgentManager::getInstance()->unloadAllPlugins();
+#endif
 }
 
 //if you want a different context,just modify the value of glContextAttrs
@@ -31,11 +42,36 @@
 
 bool AppDelegate::applicationDidFinishLaunching()
 {
+#if (CC_TARGET_PLATFORM == CC_PLATFORM_ANDROID || CC_TARGET_PLATFORM == CC_PLATFORM_IOS)
+    /**
+     * appKey、appSecret、privateKey需要从打包工具中游戏管理界面获取，替换
+     * oauthLoginServer参数是游戏服务提供的用来做登陆验证转发的接口地址。
+     */
+    std::string oauthLoginServer = "OAUTH_LOGIN_SERVER";
+    std::string appKey = "APP_KEY";
+    std::string appSecret = "APP_SERCRET";
+    std::string privateKey = "PRIVATE_KEY";
+    
+    AgentManager* pAgent = AgentManager::getInstance();
+    pAgent->init(appKey,appSecret,privateKey,oauthLoginServer);
+    
+    //使用框架中代理类进行插件初始化
+    pAgent->loadAllPlugins();
+#endif
+
     auto engine = LuaEngine::getInstance();
     ScriptEngineManager::getInstance()->setScriptEngine(engine);
     lua_State* L = engine->getLuaStack()->getLuaState();
     lua_module_register(L);
 
+#if (CC_TARGET_PLATFORM == CC_PLATFORM_ANDROID || CC_TARGET_PLATFORM == CC_PLATFORM_IOS)
+    LuaStack* stack = engine->getLuaStack();
+    lua_getglobal(stack->getLuaState(), "_G");
+    tolua_anysdk_open(stack->getLuaState());
+    tolua_anysdk_manual_open(stack->getLuaState());
+    lua_pop(stack->getLuaState(), 1);
+#endif
+
     // If you want to use Quick-Cocos2d-X, please uncomment below code
     // register_all_quick_manual(L);
 
