--- main.cpp	2015-03-13 05:25:46.000000000 +0800
+++ main.cpp	2015-10-19 11:47:00.000000000 +0800
@@ -4,15 +4,26 @@
 #include <jni.h>
 #include <android/log.h>
 #include "ide-support/SimpleConfigParser.h"
+#if (CC_TARGET_PLATFORM == CC_PLATFORM_ANDROID) && (CC_TARGET_PLATFORM == CC_PLATFORM_IOS)
+#include "PluginJniHelper.h"
+#endif
 
 #define  LOG_TAG    "main"
 #define  LOGD(...)  __android_log_print(ANDROID_LOG_DEBUG,LOG_TAG,__VA_ARGS__)
 
 using namespace cocos2d;
+#if (CC_TARGET_PLATFORM == CC_PLATFORM_ANDROID) && (CC_TARGET_PLATFORM == CC_PLATFORM_IOS)
+using namespace anysdk::framework;
+#endif
 
 void cocos_android_app_init (JNIEnv* env, jobject thiz) {
     LOGD("cocos_android_app_init");
     AppDelegate *pAppDelegate = new AppDelegate();
+#if (CC_TARGET_PLATFORM == CC_PLATFORM_ANDROID) && (CC_TARGET_PLATFORM == CC_PLATFORM_IOS)
+    JavaVM* vm;
+    env->GetJavaVM(&vm);
+    PluginJniHelper::setJavaVM(vm);
+#endif
 }
 
 extern "C"
