--- $(ANDROID_ACTIVITY_NAME).java	2013-06-11 11:51:24.000000000 +0800
+++ $(ANDROID_ACTIVITY_NAME).java	2015-11-02 14:57:40.000000000 +0800
@@ -26,14 +26,61 @@
 import org.cocos2dx.lib.Cocos2dxActivity;
 
 import android.os.Bundle;
+import org.cocos2dx.lib.Cocos2dxGLSurfaceView;
+import com.anysdk.framework.PluginWrapper;
+import android.content.Intent;
 
 public class $(ANDROID_ACTIVITY_NAME) extends Cocos2dxActivity{
-	
-	protected void onCreate(Bundle savedInstanceState){
-		super.onCreate(savedInstanceState);
-	}
-	
+    
+    protected void onCreate(Bundle savedInstanceState){
+        super.onCreate(savedInstanceState);
+        PluginWrapper.init(this);
+        PluginWrapper.setGLSurfaceView(Cocos2dxGLSurfaceView.getInstance());
+    }
+    
     static {
         System.loadLibrary("cocos2djs");
     }
+    
+    @Override
+    protected void onResume() {
+        super.onResume();
+        PluginWrapper.onResume();
+    }
+    
+    @Override
+    protected void onPause() {
+        super.onPause();
+        PluginWrapper.onPause();
+    }
+    
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+        PluginWrapper.onActivityResult(requestCode, resultCode, data);
+    }
+    
+    @Override
+    protected void onDestroy() {
+        super.onDestroy();
+        PluginWrapper.onDestroy();
+    }
+    
+    @Override
+    protected void onNewIntent(Intent intent) {
+        super.onNewIntent(intent);
+        PluginWrapper.onNewIntent(intent);
+    }
+    
+    @Override
+    protected void onRestart() {
+        super.onRestart();
+        PluginWrapper.onRestart();
+    }
+    
+    @Override
+    protected void onStop() {
+        super.onStop();
+        PluginWrapper.onStop();
+    }
 }
